src/k8s/pkg/k8sd/features/cilium/network.go

```diff
@@ -222,13 +212,6 @@ func ApplyNetwork(ctx context.Context, snap snap.Snap, s state.State, apiserver
 		"tunnelPort": config.tunnelPort,
 	}
 
-	// If we are deploying with IPv6 only, we need to set the routing mode to native
-	if ipv4CIDR == "" && ipv6CIDR != "" {
-		values["routingMode"] = "native"
```

Comment:
I believe we'll need to set this explicitly to `vxlan` since we have `--reuse-values` configured for the helm client, so that existing clusters can make the switch as well.